src/Elm/Parser/TypeAnnotation.elm

```diff
@@ -41,27 +36,34 @@ arrowRightToTypeAnnotation =
         |> Combine.continueWith typeAnnotation
 
 
-typeAnnotationNonGreedy : Parser State (Node TypeAnnotation)
-typeAnnotationNonGreedy =
+typeAnnotationNoFnExcludingTypedWithArguments : Parser State (Node TypeAnnotation)
+typeAnnotationNoFnExcludingTypedWithArguments =
     Combine.oneOf
         [ parensTypeAnnotation
-        , typedTypeAnnotation Lazy
+        , typedTypeAnnotationWithoutArguments
         , genericTypeAnnotation
         , recordTypeAnnotation
         ]
 
 
-typeAnnotationNoFn : Mode -> Parser State (Node TypeAnnotation)
-typeAnnotationNoFn mode =
-    Combine.lazy
-        (\() ->
-            Combine.oneOf
-                [ parensTypeAnnotation
-                , typedTypeAnnotation mode
-                , genericTypeAnnotation
-                , recordTypeAnnotation
-                ]
-        )
+typeAnnotationNoFnIncludingTypedWithArguments : Parser State (Node TypeAnnotation)
+typeAnnotationNoFnIncludingTypedWithArguments =
+    Combine.oneOf
+        [ parensTypeAnnotation
+        , typedTypeAnnotationWithArguments
+        , genericTypeAnnotation
+        , recordTypeAnnotation
+        ]
+
+
+typeAnnotationNoFnIncludingTypedWithArgumentsLazy : Parser State (Node TypeAnnotation)
+typeAnnotationNoFnIncludingTypedWithArgumentsLazy =
+    Combine.lazy (\() -> typeAnnotationNoFnIncludingTypedWithArguments)
```

Comment:
We can also inline these declarations if we want to:
```elm
Combine.lazy (\() -> typeAnnotationNoFnIncludingTypedWithArguments)
  |> Combine.andThen ...
```
(Just mentioning it because I realized it now, by looking at the Elm docs)